aten/src/ATen/native/TensorFactories.cpp

```diff
@@ -59,6 +59,10 @@ void window_function_checks(
       window_length);
 }
 
+const TypeExtendedInterface& getFactoryType(const TensorOptions& options) {
+  return static_cast<const TypeExtendedInterface&>(at::getType(options));
```

Comment:
similar point here to the issue below -- this function should probably just return a TypeExtendedInterface or we should have a similar function that does that.